internal/pkg/plugin/python_plugin_installer.go

```diff
@@ -21,14 +21,18 @@ func (p *PythonPluginInstaller) CheckVersion(ver *version.Version) error {
 	versionCmd := exec.NewCommand("python", "--version")
 	version3Cmd := exec.NewCommand("python3", "--version")
 
-	v3, _ := version.NewVersion("3.0.0")
+	versionSegments := ver.Segments()
+	if len(versionSegments) == 0 {
+		return errors.New(errors.NoVersionFoundErrorMsg)
+	}
+	majorVer := versionSegments[0]
 
 	var out []byte
 	var err error
-	if ver.GreaterThanOrEqual(v3) {
+	if majorVer == 3 {
 		out, err = version3Cmd.Output()
 	}
-	if err != nil || ver.LessThan(v3) {
+	if err != nil || majorVer != 3 {
 		out, err = versionCmd.Output()
```

Comment:
The `python` binary could still be Python 3. `python3` is just a common alias for `python`.

Reply:
Aliases aren't supported with `exec.Command` unless you execute through bash or zsh depending on which `rc` file you put the alias in (also, sometimes `python` is the alias instead of `python3`).

That's why I also check `python` in the case that `python3` failed.

Reply:
> Aliases aren't supported with exec.Command

It's not an alias in the sense of the `alias` shell command, it's a real command! You should grab a `python3` binary to test this with since it sounds like this is untested...

Reply:
I am using a python3 binary.

I'm confused now about what's problematic with this if statement

Reply:
Oops, I was thinking `&&` and not `||`, disregard. Tracing the values of `err` here was hurting my head a bit...

Reply:
Ahh, gotcha.